torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -753,7 +764,7 @@ std::shared_ptr<SugaredValue> toSugaredValue(
     }
   }
 
-  py::bool_ isFunction = py::module::import("inspect").attr("isfunction")(obj);
+  py::bool_ isFunction = py::module::import("inspect").attr("isroutine")(obj);
```

Comment:
Can you explain why this was changed? I ran the following script: 
```
class Foo:
    @staticmethod
    def my_static(x, y):
        return x + y
from inspect import getattr_static

def is_static_fn(cls, fn):
    return isinstance(getattr_static(cls, fn), staticmethod)

def get_static_fn(cls, fn):
    return getattr_static(cls, fn).__func__

print(inspect.isfunction(get_static_fn(Foo, "my_static")))
> True
```
So it seems like we don't need `isroutine` to detect the static functions returned by the new helpers this PR adds. From the docs, it seems the only difference is that it includes builtins, which doesn't seem like it's that relevant?

Reply:
This is indeed not needed. It seemed to throw errors before, but it was probably something else and I got my wires crossed. Removed.